### PR TITLE
Plfm 4847

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/migration/async/BackupJobExecutorImpl.java [hunk 1]
```diff
@@ -25,20 +29,54 @@ public BackupJobExecutorImpl(Configuration configuration, AsynchronousJobExecuto
...
+	 * @param minimumId
+	 * @param maximumId
+	 * @param sparseRange
+	 * @return
+	 */
+	public static List<BackupTypeRangeRequest> createContiguousBackupRangeRequests(final BackupAliasType backupAlisType,
+			final long batchSize, final MigrationType migrationType, final long minimumId, final long maximumId,
+			List<IdRange> sparseRange) {
+		ValidateArgument.required(sparseRange, "sparseRange");
+		ValidateArgument.required(!sparseRange.isEmpty(), "sparseRange is empty");
```
Comment (xschildw) [inline src/main/java/org/sagebionetworks/migration/async/BackupJobExecutorImpl.java]: Could be empty if no values in range

File: src/test/java/org/sagebionetworks/migration/async/BackupJobExecutorImplTest.java [hunk 2]
```diff
@@ -15,60 +18,174 @@
...
+		assertEquals(1, results.size());
+		// one
+		BackupTypeRangeRequest request = results.get(0);
+		assertEquals(backupAliasType, request.getAliasType());
+		assertEquals(new Long(batchSize), request.getBatchSize());
+		assertEquals(migrationType, request.getMigrationType());
+		assertEquals(new Long(1), request.getMinimumId());
+		assertEquals(new Long(8), request.getMaximumId());
+	}
+
```
Comment (xschildw) [inline src/test/java/org/sagebionetworks/migration/async/BackupJobExecutorImplTest.java:171]: Add test for empty range result from server
